File: framework/languages/java/zlink-framework-core/src/main/java/systems/zlink/framework/actors/ZLinkActor.java
package systems.zlink.framework.actors;

import java.util.concurrent.CompletableFuture;
import java.util.concurrent.CompletionStage;

public interface ZLinkActor {
    ZLinkActorContext context();

    default void configure() {
    }

    default CompletionStage<Void> onJoinCompleted(
        ZLinkActorJoinCompletion completion) {
        return CompletableFuture.completedFuture(null);
    }
}

File: framework/languages/node/packages/framework/src/contracts/Actors/ZLinkActor.ts
import type { ZLinkActorContext } from './ZLinkActorContext';
import type { ZLinkActorJoinCompletion } from './ZLinkActorFactory';

export interface ZLinkActor {
  readonly context: ZLinkActorContext;
  configure?(): void;
  onJoinCompleted?(completion: ZLinkActorJoinCompletion): Promise<void>;
}
